app/common/model/src/main/java/io/syndesis/common/model/DataShape.java

```diff
@@ -90,5 +99,94 @@
     }
 
     class Builder extends ImmutableDataShape.Builder {
+
+        public static final String COMPRESSION_METADATA_KEY = "compression";
+        private static final String COMPRESSED_METADATA_KEY = "compressed";
+
+        public Builder decompress() {
+            DataShape currentShape = build();
+            if (Objects.equals(currentShape.getMetadata().get(COMPRESSION_METADATA_KEY), "true") && Objects.equals(currentShape.getMetadata().getOrDefault(COMPRESSED_METADATA_KEY, "true"), "true")) {
```

Comment:
It is checked for COMPRESSION_METADATA_KEY and COMPRESSED_METADATA_KEY where the first is for compression enabled at all and the 2nd to mark that compression has already been performed in order to avoid multiple compressions.

However, I will simplify the predicate a bit